packages/SwingSet/src/vats/comms/delivery.js

```diff
@@ -273,74 +305,73 @@ export function makeDeliveryKit(state, syscall, transmit, clistKit, stateKit) {
     transmit(remoteID, msg);
   }
 
-  function handleResolution(vpid, resolution) {
-    // resolution: { type, slot or data }
-    insistVatType('promise', vpid);
-    insistPromiseIsUnresolved(vpid);
-    insistDeciderIsComms(vpid);
-
-    const { subscribers, kernelIsSubscribed } = getPromiseSubscribers(vpid);
-
-    // mark it as resolved in the promise table, so later messages to it will
-    // be handled properly
-    markPromiseAsResolved(vpid, resolution);
+  function handleResolutions(resolutions) {
+    const [[primaryVpid]] = resolutions;
+    const { subscribers, kernelIsSubscribed } = getPromiseSubscribers(
```

Comment:
This feels wonky.. I think we'll need to make sure subscribers of all the promises are notified, not just the subscribers of the first one. I believe it's possible for one of the secondary promises in the batch to be one that the vat already knows about:

* send P1 (unresolved) out through comms vat to remote A
* send P2 (unresolved) out through comms vat to remote B
* resolve P1 to data that contains P2
  * now the `notify` event is on the run-queue, but has not yet reached the comms vat
* resolve P2 to something
  * now there are two `notify` events on the run-queue
* the `notify(comms-vat, P1)` event reaches the top
  * the kernel creates a batch with P1 as the first promise, and P2 as the second
  * comms needs to send a two-promise batch (P1+P2) to remote A
  * comms needs to send a one-promise batch (just P2) to remote B
  * comms retires the kernel-facing vrefs for both P1 and P2
* the `notify(comms-vat, P2)` event reaches the top
  * kernel ignores it because the P2 c-list entry was already retired

Figuring out what batches to create seems non-trivial, though, let's talk this through. I imagine this PR is good to land for now, but we'll need to examine this more closely and refine it in a later PR to make sure we're getting all the cases right.

Reply:
That sounds like a plausible argument, but I think that would explode into a bunch of very complicated mechanism which I'm not sure it is needed here.  Keep in mind that `primaryVpid` is the root of a reference graph and that the comms vat is a simple pass through.  So when `notify` notifies a bundle, that bundle travels as a unit through the network to the recipients; all the resolved promises in the bundle are in the same place.

If my thinking here is wrong, then we may have a more complicated mess on our hands.

Reply:
I have filed ticket #2249 to make sure we don't lose track of this.